### DISCUSSION TITLE
Python API

### DISCUSSION THREAD

**Discussion by themarkjensen** [bootstrap-email/bootstrap-email#99] [Ideas]
This is a great tool; thanks for making it!

I'm currently using the online editor, but I think this would be excellent as a python library. This would work very well alongside the Jinja2 templating engine so that emails could be written in HTML with bootstrap classes, compiled using the python api, and then rendered using the Jinja2 engine, and finally sent off to your favorite email provider (e.g. sendgrid, mailchimp).

e.g.
```python
from bootstrap-email import BootstrapEmail
from flask import render_template

be = BootstrapEmail()

with open('template_written_in_bootstrap.html', 'r') as file:
    template_written_in_bootstrap = file.read()

compiled_template = be.compile(template_written_in_bootstrap)

# Code to get other objects to pass into rendering engine
user = get_user()
products = get_a_list_of_products()

rendered_email = render_template(user=user, products=products)

# Send email using your choice of email provider
...

```

**Comment by stuyam** [comment]
Support for more languages would be amazing. However currently the best thing I can direct you to is the CLI implementation. It would be the best approach for something like that where you want it to be programmatic and not have to use the editor. Some people use it as part of their build process or run in manually. https://v1.bootstrapemail.com/docs/setup#command-line
